build_tools/cmake/iree_static_linker_test.cmake

```diff
@@ -0,0 +1,259 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(CMakeParseArguments)
```

Comment:
Oh... I think we can actually remove these includes? (separate cleanup from this PR)

https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html
> _New in version 3.5:_ This command is implemented natively. Previously, it has been defined in the module [CMakeParseArguments](https://cmake.org/cmake/help/latest/module/CMakeParseArguments.html#module:CMakeParseArguments).

https://cmake.org/cmake/help/latest/module/CMakeParseArguments.html
> This module once implemented the [cmake_parse_arguments()](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html#command:cmake_parse_arguments) command that is now implemented natively by CMake. It is now an empty placeholder for compatibility with projects that include it to get the command from CMake 3.4 and lower.

Reply:
Removed.